## Libraries for reading in data

In [137]:
import pandas as pd
import os
import numpy as np
notebook_directory = os.getcwd()
print(f'Current notebook directory: {notebook_directory}')

Current notebook directory: c:\Users\Matth\Desktop\repos\CollabProject


## __Dataset 1__: `NBA-BoxScores-2024-2025.csv`
* 736 NBA player-games from the 2024-2025 season.
*  Contains 440 players.

In [138]:
box_score_data_path = os.path.join(notebook_directory,'data','NBA-BoxScores-2024-2025.csv')
df_boxscores = pd.read_csv(box_score_data_path)
np.shape(df_boxscores)

(736, 30)

In [139]:
df_boxscores.sample(10) # looks like this consists of player-games of the 2024 to 2025 NBA seasons

Unnamed: 0   GAME_ID     TEAM_ID TEAM_ABBREVIATION      TEAM_CITY  \
224           3  22400065  1610612753               ORL        Orlando   
60            7  22400072  1610612744               GSW   Golden State   
79           26  22400072  1610612757               POR       Portland   
396          28  22400075  1610612743               DEN         Denver   
166           1  22400067  1610612739               CLE      Cleveland   
732          24  22400086  1610612757               POR       Portland   
373           5  22400075  1610612760               OKC  Oklahoma City   
609          13  22400082  1610612745               HOU        Houston   
467          11  22400077  1610612751               BKN       Brooklyn   
597           1  22400082  1610612763               MEM        Memphis   

     PLAYER_ID               PLAYER_NAME    NICKNAME START_POSITION  \
224     203484  Kentavious Caldwell-Pope  Kentavious              G   
60     1641764        Brandin Podziemski     Brandin            NaN   
79     1631101            Shaedon Sharpe     Shaedon            NaN   
396     201599            DeAndre Jordan     DeAndre            NaN   
166    1630596               Evan Mobley        Evan              F   
732    1641712              Rayan Rupert       Rayan            NaN   
373    1641717             Cason Wallace       Cason            NaN   
609    1631095          Jabari Smith Jr.      Jabari              F   
467    1641798             Jaylen Martin      Jaylen            NaN   
597    1630583              Santi Aldama       Santi              F   

                    COMMENT  ... OREB  DREB   REB  AST  STL  BLK   TO   PF  \
224                     NaN  ...  1.0   0.0   1.0  2.0  1.0  2.0  2.0  1.0   
60                      NaN  ...  4.0   3.0   7.0  4.0  1.0  0.0  3.0  3.0   
79     DND - Injury/Illness  ...  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN   
396  DNP - Coach's Decision  ...  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN   
166                     NaN  ...  4.0   5.0   9.0  3.0  2.0  3.0  1.0  2.0   
732  DNP - Coach's Decision  ...  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN   
373                     NaN  ...  2.0   2.0   4.0  0.0  2.0  0.0  0.0  1.0   
609                     NaN  ...  6.0  10.0  16.0  4.0  0.0  2.0  3.0  5.0   
467  DNP - Coach's Decision  ...  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN   
597                     NaN  ...  0.0   5.0   5.0  3.0  0.0  2.0  2.0  2.0   

      PTS  PLUS_MINUS  
224   3.0        24.0  
60    0.0        35.0  
79    NaN         NaN  
396   NaN         NaN  
166  25.0        31.0  
732   NaN         NaN  
373   4.0        18.0  
609  14.0        18.0  
467   NaN         NaN  
597   2.0       -13.0  

[10 rows x 30 columns]

# __Dataset 2__: `2025teamstats.csv`
* NBA team stats for the 2024-2025 season.

In [140]:
team_stats2_data_path = os.path.join(notebook_directory,'data','2025teamstats.csv')
df_teamstats2025 = pd.read_csv(team_stats2_data_path)
np.shape(df_teamstats2025)

(31, 25)

In [141]:
df_teamstats2025.sample(10)

Rk                    Team   G     MP    FG   FGA    FG%    3P   3PA  \
15  16.0         Houston Rockets  57  241.3  41.6  93.2  0.447  12.2  35.5   
17  18.0   Golden State Warriors  57  240.4  41.0  91.8  0.446  15.3  42.5   
14  15.0      Los Angeles Lakers  55  240.5  41.3  85.5  0.482  12.4  34.6   
13  14.0            Phoenix Suns  57  241.8  40.7  85.8  0.474  13.9  37.4   
16  17.0       San Antonio Spurs  55  240.9  41.1  89.5  0.460  13.8  39.5   
0    1.0     Cleveland Cavaliers  57  240.0  45.1  90.6  0.498  16.1  41.2   
7    8.0        Sacramento Kings  56  243.1  43.3  91.2  0.474  12.4  35.8   
30   NaN          League Average  57  241.3  41.5  89.2  0.465  13.4  37.5   
24  25.0  Portland Trail Blazers  57  240.4  40.1  88.7  0.453  12.5  35.9   
20  21.0    Los Angeles Clippers  56  241.3  40.8  86.6  0.471  12.2  33.9   

      3P%  ...    FT%   ORB   DRB   TRB   AST  STL  BLK   TOV    PF    PTS  
15  0.343  ...  0.747  14.5  33.4  47.9  22.8  8.7  5.1  13.8  19.8  112.9  
17  0.360  ...  0.737  12.8  33.2  45.9  28.9  8.9  5.0  13.8  19.5  112.5  
14  0.357  ...  0.793   9.3  32.9  42.1  26.7  7.7  4.8  13.8  16.7  113.1  
13  0.373  ...  0.805   9.1  33.3  42.4  27.4  7.5  5.1  14.0  18.1  113.2  
16  0.349  ...  0.783  10.9  33.8  44.7  28.9  8.1  6.5  14.1  16.5  112.7  
0   0.391  ...  0.783  10.8  34.4  45.2  29.1  8.4  4.4  13.2  18.6  122.9  
7   0.345  ...  0.806  11.1  33.8  44.9  26.9  7.8  4.6  13.5  19.2  116.6  
30  0.359  ...  0.780  11.1  33.1  44.2  26.4  8.3  5.0  14.5  18.9  113.4  
24  0.347  ...  0.772  12.4  31.3  43.6  23.6  8.1  5.6  16.0  19.5  109.6  
20  0.359  ...  0.797  10.6  33.4  44.0  24.7  9.8  4.5  15.8  18.5  110.9  

[10 rows x 25 columns]

# __Dataset 3__: `TeamStatistics.csv`
* Data from all NBA games.
* From 1946.

In [142]:
# dataset - https://www.kaggle.com/datasets/eoinamoore/historical-nba-data-and-player-box-scores/
games_big_data_path = os.path.join(notebook_directory,'data','TeamStatistics.csv')
df_games_big = pd.read_csv(games_big_data_path)
np.shape(df_games_big)

(142836, 48)

In [143]:
df_games_big.sample(7) # quick look at the data

gameId             gameDate      teamCity   teamName      teamId  \
50023   20600468  2007-01-03 19:00:00    Washington    Wizards  1610612764   
61877   20200005  2002-10-30 19:00:00  Philadelphia      76ers  1610612755   
13017   21900970  2020-03-11 19:30:00     Charlotte    Hornets  1610612766   
50843   20600055  2006-11-07 20:00:00       Memphis  Grizzlies  1610612763   
4813    22201218  2023-04-09 13:00:00     Charlotte    Hornets  1610612766   
6410    22200421  2022-12-14 21:00:00     Cleveland  Cavaliers  1610612739   
112770  27900077  1979-10-26 20:00:00       Atlanta      Hawks  1610612737   

       opponentTeamCity opponentTeamName  opponentTeamId  home  win  ...  \
50023         Milwaukee            Bucks      1610612749     1    1  ...   
61877         Milwaukee            Bucks      1610612749     1    1  ...   
13017             Miami             Heat      1610612748     0    1  ...   
50843           Houston          Rockets      1610612745     1    0  ...   
4813          Cleveland        Cavaliers      1610612739     0    1  ...   
6410             Dallas        Mavericks      1610612742     0    1  ...   
112770       New Jersey             Nets      1610612751     0    1  ...   

        leadChanges  pointsFastBreak  pointsFromTurnovers  pointsInThePaint  \
50023           NaN              NaN                  NaN               NaN   
61877           NaN              NaN                  NaN               NaN   
13017           NaN              NaN                  NaN               NaN   
50843           NaN              NaN                  NaN               NaN   
4813            NaN              NaN                  NaN               NaN   
6410            NaN              NaN                  NaN               NaN   
112770          NaN              NaN                  NaN               NaN   

        pointsSecondChance  timesTied  timeoutsRemaining  seasonWins  \
50023                  NaN        NaN                NaN         NaN   
61877                  NaN        NaN                NaN         NaN   
13017                  NaN        NaN                NaN         NaN   
50843                  NaN        NaN                NaN         NaN   
4813                   NaN        NaN                NaN         NaN   
6410                   NaN        NaN                NaN         NaN   
112770                 NaN        NaN                NaN         NaN   

        seasonLosses  coachId  
50023            NaN      NaN  
61877            NaN      NaN  
13017            NaN      NaN  
50843            NaN      NaN  
4813             NaN      NaN  
6410             NaN      NaN  
112770           NaN      NaN  

[7 rows x 48 columns]

In [144]:
df_games_big.coachId.unique() # nothing in coach id column. We'll get rid of it.
df_games_big = df_games_big.drop(columns=['coachId'])

In [145]:
# checking what is redundant across rows of same game
df_games_big.loc[2240:2241,].T # we should check for NaNs

2240                 2241
gameId                              22301095             22301095
gameDate                 2024-04-02 19:00:00  2024-04-02 19:00:00
teamCity                           Milwaukee           Washington
teamName                               Bucks              Wizards
teamId                            1610612749           1610612764
opponentTeamCity                  Washington            Milwaukee
opponentTeamName                     Wizards                Bucks
opponentTeamId                    1610612764           1610612749
home                                       0                    1
win                                        0                    1
teamScore                                113                  117
opponentScore                            117                  113
assists                                 31.0                 26.0
blocks                                   6.0                  2.0
steals                                   8.0                  9.0
fieldGoalsAttempted                    103.0                 93.0
fieldGoalsMade                          45.0                 47.0
fieldGoalsPercentage                   0.437                0.505
threePointersAttempted                  48.0                 38.0
threePointersMade                       11.0                  9.0
threePointersPercentage                0.229                0.237
freeThrowsAttempted                     18.0                 17.0
freeThrowsMade                          12.0                 14.0
freeThrowsPercentage                   0.667                0.824
reboundsDefensive                       39.0                 39.0
reboundsOffensive                       18.0                  5.0
reboundsTotal                           57.0                 44.0
foulsPersonal                           17.0                 20.0
turnovers                               12.0                 11.0
plusMinusPoints                         -4.0                  4.0
numMinutes                             240.0                240.0
q1Points                                 NaN                  NaN
q2Points                                 NaN                  NaN
q3Points                                 NaN                  NaN
q4Points                                 NaN                  NaN
benchPoints                              NaN                  NaN
biggestLead                              NaN                  NaN
biggestScoringRun                        NaN                  NaN
leadChanges                              NaN                  NaN
pointsFastBreak                          NaN                  NaN
pointsFromTurnovers                      NaN                  NaN
pointsInThePaint                         NaN                  NaN
pointsSecondChance                       NaN                  NaN
timesTied                                NaN                  NaN
timeoutsRemaining                        NaN                  NaN
seasonWins                               NaN                  NaN
seasonLosses                             NaN                  NaN

Missingness of df_teams

In [146]:
# missing percentage per column
cols_without_missing = []
for col in df_games_big.columns:
    missing_percent = df_games_big[col].isna().sum()/len(df_games_big)
    if missing_percent > 0.30: # if more than 30% of the column is missing, we want to drop it
        cols_without_missing.append(False)
    else: 
        cols_without_missing.append(True)

# removing the columns with too many missing values
df_games_big = df_games_big[df_games_big.columns[cols_without_missing]]

# __Dataset 4__: `Games.csv`
* This data also contains information on every NBA game. 
* Each row is a game, which specifies the home and away teams
* This is different than the previous dataset, which has two rows for each game.

In [147]:
games_small_data_path = os.path.join(notebook_directory,'data','Games.csv')
df_games_small = pd.read_csv(games_small_data_path)
np.shape(df_games_small)

C:\Users\Matth\AppData\Local\Temp\ipykernel_28124\4217729876.py:2: DtypeWarning: Columns (15) have mixed types. Specify dtype option on import or set low_memory=False.
  df_games_small = pd.read_csv(games_small_data_path)


(71418, 17)

In [148]:
df_games_small.sample(7)

gameId             gameDate hometeamCity   hometeamName  hometeamId  \
29321  20300331  2003-12-14 19:00:00     New York         Knicks  1610612752   
51701  28300847  1984-03-30 20:00:00   Washington        Bullets  1610612764   
7588   41800165  2019-04-23 22:30:00     Portland  Trail Blazers  1610612757   
49438  28600065  1986-11-11 20:00:00      Chicago          Bulls  1610612741   
43757  29100265  1991-12-11 20:00:00    Charlotte        Hornets  1610612766   
11270  21600340  2016-12-09 20:00:00    Milwaukee          Bucks  1610612749   
37949  29600092  1996-11-13 19:30:00    Cleveland      Cavaliers  1610612739   

        awayteamCity   awayteamName  awayteamId  homeScore  awayScore  \
29321     Washington        Wizards  1610612764         89         87   
51701       New York         Knicks  1610612752        107         79   
7588   Oklahoma City        Thunder  1610612760        118        115   
49438        Atlanta          Hawks  1610612737        112        110   
43757         Dallas      Mavericks  1610612742        103        106   
11270        Atlanta          Hawks  1610612737        110        114   
37949       Portland  Trail Blazers  1610612757         81         70   

           winner        gameType  attendance  arenaId           gameLabel  \
29321  1610612752  Regular Season     17191.0       30                 NaN   
51701  1610612764  Regular Season      9255.0       74                 NaN   
7588   1610612757        Playoffs     20241.0       51  West - First Round   
49438  1610612741  Regular Season     16877.0       66                 NaN   
43757  1610612742  Regular Season     23698.0        9                 NaN   
11270  1610612737  Regular Season     16289.0        5                 NaN   
37949  1610612739  Regular Season     13757.0       22                 NaN   

      gameSubLabel  seriesGameNumber  
29321          NaN               NaN  
51701          NaN               NaN  
7588           NaN               5.0  
49438          NaN               NaN  
43757          NaN               NaN  
11270          NaN               NaN  
37949          NaN               NaN

# Consolidating Datasets 3 and 4
* Dataset 3: `TeamStatistics.csv` has 142,836 rows, 2 per game
* Dataset 4: `Games.csv` has 71,418 rows, 1 per game

In [149]:
# both datasets with games have a game ID and a game date
print(f"Shared columns:\n\n\t{df_games_small.columns[df_games_small.columns.isin(df_games_big.columns)]}\n\n") 

# columns of df_games_big NOT in df_games
print(f"Columns of big df not in smaller:\n\n\t{df_games_big.columns[~df_games_big.columns.isin(df_games_small.columns)]}\n\n")

 # columns of df_games NOT in df_games_big
print(f"Columns of small df not in big:\n\n\t{df_games_small.columns[~df_games_small.columns.isin(df_games_big.columns)]}\n\n")

Shared columns:

	Index(['gameId', 'gameDate'], dtype='object')


Columns of big df not in smaller:

	Index(['teamCity', 'teamName', 'teamId', 'opponentTeamCity',
       'opponentTeamName', 'opponentTeamId', 'home', 'win', 'teamScore',
       'opponentScore', 'assists', 'blocks', 'steals', 'fieldGoalsAttempted',
       'fieldGoalsMade', 'fieldGoalsPercentage', 'threePointersAttempted',
       'threePointersMade', 'threePointersPercentage', 'freeThrowsAttempted',
       'freeThrowsMade', 'freeThrowsPercentage', 'reboundsDefensive',
       'reboundsOffensive', 'reboundsTotal', 'foulsPersonal', 'turnovers',
       'plusMinusPoints', 'numMinutes'],
      dtype='object')


Columns of small df not in big:

	Index(['hometeamCity', 'hometeamName', 'hometeamId', 'awayteamCity',
       'awayteamName', 'awayteamId', 'homeScore', 'awayScore', 'winner',
       'gameType', 'attendance', 'arenaId', 'gameLabel', 'gameSubLabel',
       'seriesGameNumber'],
      dtype='object')




In [150]:
# subsetting by home teams
df_games_big_home = df_games_big[df_games_big.home == 1]
print(f"Home teams data size: {df_games_big_home.shape}")

# subsetting by away teams
df_games_big_away = df_games_big[df_games_big.home == 0]
print(f"Away teams data size: {df_games_big_away.shape}")


Home teams data size: (71418, 31)
Away teams data size: (71418, 31)


In [151]:
# merging home and away teams dataframes
df_games_wide = df_games_big_home.merge(right=df_games_big_away,how='inner',on='gameId',suffixes=('_home','_away'))
print(df_games_wide.shape)
df_games_wide.sample(6)


(71418, 61)


gameId        gameDate_home teamCity_home teamName_home  teamId_home  \
2016   22300048  2023-11-24 20:00:00       Houston       Rockets   1610612745   
7226   21900251  2019-11-26 20:30:00        Dallas     Mavericks   1610612742   
44739  29000458  1991-01-12 20:00:00    Washington       Bullets   1610612764   
59438  27500679  1976-03-31 19:35:00       Detroit       Pistons   1610612765   
9986   21700248  2017-11-20 20:00:00   New Orleans      Pelicans   1610612740   
26741  20500166  2005-11-23 22:30:00  Golden State      Warriors   1610612744   

      opponentTeamCity_home opponentTeamName_home  opponentTeamId_home  \
2016                 Denver               Nuggets           1610612743   
7226            Los Angeles              Clippers           1610612746   
44739                Boston               Celtics           1610612738   
59438               Chicago                 Bulls           1610612741   
9986          Oklahoma City               Thunder           1610612760   
26741           San Antonio                 Spurs           1610612759   

       home_home  win_home  ...  freeThrowsAttempted_away  \
2016           1         1  ...                      21.0   
7226           1         0  ...                      30.0   
44739          1         1  ...                      21.0   
59438          1         1  ...                      34.0   
9986           1         1  ...                      26.0   
26741          1         0  ...                      20.0   

       freeThrowsMade_away  freeThrowsPercentage_away  reboundsDefensive_away  \
2016                  15.0                      0.714                    30.0   
7226                  21.0                      0.700                    39.0   
44739                 21.0                      1.000                    22.0   
59438                 28.0                      0.824                     NaN   
9986                  20.0                      0.769                    31.0   
26741                 17.0                      0.850                    39.0   

       reboundsOffensive_away  reboundsTotal_away  foulsPersonal_away  \
2016                     16.0                46.0                14.0   
7226                     17.0                56.0                27.0   
44739                    11.0                33.0                20.0   
59438                     NaN                 NaN                27.0   
9986                      9.0                40.0                25.0   
26741                    13.0                52.0                23.0   

       turnovers_away  plusMinusPoints_away  numMinutes_away  
2016             15.0                 -19.0            240.0  
7226             14.0                  15.0            240.0  
44739            24.0                 -17.0            240.0  
59438             NaN                  -6.0            240.0  
9986             14.0                  -7.0            240.0  
26741            14.0                  24.0            240.0  

[6 rows x 61 columns]

In [152]:
# merging df_games and home and away teams (df_games_wide) dataframes to make one big df_games dataframe
df_games = df_games_small.merge(right=df_games_wide,how='inner',on='gameId',suffixes=('_teams','_games'))
print(df_games.shape)
df_games.sample(7)

(71418, 77)


gameId             gameDate  hometeamCity hometeamName  hometeamId  \
11498  21600113  2016-11-09 21:00:00       Phoenix         Suns  1610612756   
70224  25300320  1954-03-13 19:00:00      New York       Knicks  1610612752   
11179  21600432  2016-12-21 20:00:00       Chicago        Bulls  1610612741   
11717  41500315  2016-05-26 21:00:00  Golden State     Warriors  1610612744   
26646  20500258  2005-12-07 19:00:00       Toronto      Raptors  1610612761   
22610  20800043  2008-11-03 19:00:00       Orlando        Magic  1610612753   
30581  20200361  2002-12-18 21:00:00       Houston      Rockets  1610612745   

        awayteamCity awayteamName  awayteamId  homeScore  awayScore  ...  \
11498        Detroit      Pistons  1610612765        107        100  ...   
70224       Syracuse    Nationals  1610612755         67         60  ...   
11179     Washington      Wizards  1610612764         97        107  ...   
11717  Oklahoma City      Thunder  1610612760        120        111  ...   
26646    Los Angeles       Lakers  1610612747         91        102  ...   
22610        Chicago        Bulls  1610612741         96         93  ...   
30581        Indiana       Pacers  1610612754         95         83  ...   

       freeThrowsAttempted_away freeThrowsMade_away  \
11498                      32.0                27.0   
70224                       NaN                18.0   
11179                      25.0                18.0   
11717                      24.0                20.0   
26646                      15.0                 9.0   
22610                      18.0                12.0   
30581                      32.0                23.0   

       freeThrowsPercentage_away  reboundsDefensive_away  \
11498                      0.844                    32.0   
70224                        NaN                     NaN   
11179                      0.720                    38.0   
11717                      0.833                    32.0   
26646                      0.600                    38.0   
22610                      0.667                    35.0   
30581                      0.719                    25.0   

      reboundsOffensive_away reboundsTotal_away  foulsPersonal_away  \
11498                    9.0               41.0                24.0   
70224                    NaN                NaN                23.0   
11179                    6.0               44.0                20.0   
11717                   13.0               45.0                26.0   
26646                   10.0               48.0                25.0   
22610                   11.0               46.0                26.0   
30581                    9.0               34.0                24.0   

      turnovers_away plusMinusPoints_away numMinutes_away  
11498           12.0                 -7.0           240.0  
70224            NaN                 -7.0             0.0  
11179           10.0                 10.0           240.0  
11717           17.0                 -9.0           240.0  
26646           19.0                 11.0           240.0  
22610           18.0                 -3.0           240.0  
30581           14.0                -12.0           240.0  

[7 rows x 77 columns]

In [ ]:
print(df_games.columns.tolist())
# columns to drop

redundant_cols = ['hometeamCity',
                  'awayteamCity',
                  'hometeamId',
                  'awayteamCity',
                  'awayteamId',
                  'gameDate_home',
                  'teamCity_home',
                  'teamName_home',
                  'teamId_home',
                  'opponentTeamCity_home',
                  'opponentTeamName_home',
                  'opponentTeamId_home',
                  'home_home',
                  'teamScore_home',
                  ''

                  ]

['gameId', 'gameDate', 'hometeamCity', 'hometeamName', 'hometeamId', 'awayteamCity', 'awayteamName', 'awayteamId', 'homeScore', 'awayScore', 'winner', 'gameType', 'attendance', 'arenaId', 'gameLabel', 'gameSubLabel', 'seriesGameNumber', 'gameDate_home', 'teamCity_home', 'teamName_home', 'teamId_home', 'opponentTeamCity_home', 'opponentTeamName_home', 'opponentTeamId_home', 'home_home', 'win_home', 'teamScore_home', 'opponentScore_home', 'assists_home', 'blocks_home', 'steals_home', 'fieldGoalsAttempted_home', 'fieldGoalsMade_home', 'fieldGoalsPercentage_home', 'threePointersAttempted_home', 'threePointersMade_home', 'threePointersPercentage_home', 'freeThrowsAttempted_home', 'freeThrowsMade_home', 'freeThrowsPercentage_home', 'reboundsDefensive_home', 'reboundsOffensive_home', 'reboundsTotal_home', 'foulsPersonal_home', 'turnovers_home', 'plusMinusPoints_home', 'numMinutes_home', 'gameDate_away', 'teamCity_away', 'teamName_away', 'teamId_away', 'opponentTeamCity_away', 'opponentTeamNam

In [ ]:
import pandas as pda
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn
import warnings

from sklearn.preprocessing import LabelEncoder
from sklearn.impute import KNNImputer
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import f1_score
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import cross_val_score


In [ ]:
team1 = input("Enter Team 1:")
team2 = input("Enter Team 2:")

team1_df = df_teams[(df_teams['teamName'] == team1) & (df_teams['opponentTeamName'] == team2)]
team1_df

team2_df = df_teams[(df_teams['teamName'] == team2) & (df_teams['opponentTeamName'] == team1)]
team2_df

#model for team1
features = ['assists', 'blocks', 'steals', 'fieldGoalsMade', 'fieldGoalsPercentage', 'threePointersMade', 'freeThrowsMade', 'reboundsTotal', 'turnovers']
x = team1_df[features]
y = team1_df['teamScore']

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)
model = RandomForestRegressor(n_estimators=100, random_state=42)
model.fit(x_train, y_train)

#model for team2
features2 = ['assists', 'blocks', 'steals', 'fieldGoalsMade', 'fieldGoalsPercentage', 'threePointersMade', 'freeThrowsMade', 'reboundsTotal', 'turnovers']
x = team2_df[features]
y = team2_df['teamScore']

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)
model2 = RandomForestRegressor(n_estimators=100, random_state=42)
model2.fit(x_train, y_train)

RandomForestRegressor(random_state=42)

In [ ]:
team1_score = team1_df[['assists', 'blocks', 'steals', 'fieldGoalsMade', 'fieldGoalsPercentage', 'threePointersMade', 'freeThrowsMade', 'reboundsTotal', 'turnovers']]
team2_score = team2_df[['assists', 'blocks', 'steals', 'fieldGoalsMade', 'fieldGoalsPercentage', 'threePointersMade', 'freeThrowsMade', 'reboundsTotal', 'turnovers']]

predicted_score1 = model.predict(team1_score)[0]
predicted_score2 = model2.predict(team2_score)[0]

print(f'{team1} Score: {predicted_score1}')
print(f'{team2} Score: {predicted_score2}')

Bulls Score: 107.29
Celtics Score: 122.48
